docs/docs/how-to/local-development/environment-variables.md

```diff
@@ -122,83 +40,23 @@ module.exports = {
-
-Specifically, you can use `dotenv` and your individual OS Env Var to locate the `.env.myCustomEnvironment` file, and then use `module.exports` to store those Project Env Vars somewhere that the client-side JavaScript can access the values (via GraphQL queries).
-
-### Google Analytics env var example
-
-If you would like to add a `staging` environment with a custom Google Analytics Tracking ID, and a dedicated `apiUrl` you can add `.env.staging` at the root of your project. In order to do so, use the following modification to your `gatsby-config.js`:
-
-```text:title=.env.staging
-GA_TRACKING_ID="UA-1234567890"
-API_URL="http://foo.bar"
-```
-
-```javascript:title=gatsby-config.js
-const activeEnv =
-  process.env.GATSBY_ACTIVE_ENV || process.env.NODE_ENV || "development"
+## Accessing Environment Variables in the browser.
 
-console.log(`Using environment config: '${activeEnv}'`)
+By default, environment variables are only available in Node.js code and are not available in the browser as some
+variables should be kept secret and not exposed to anyone visiting the site.
 
-require("dotenv").config({
-  path: `.env.${activeEnv}`,
-})
+To expose a variable in the browser, you must preface its name with `GATSBY_`. So `GATSBY_API_URL` will be available in
+browser code but `API_KEY` will not.
 
-module.exports = {
-  siteMetadata: {
-    title: "Gatsby Default Starter",
-    apiUrl: process.env.API_URL,
-  },
-  plugins: [
-    {
-      resolve: `gatsby-plugin-google-analytics`,
-      options: {
-        trackingId: process.env.GA_TRACKING_ID,
-        // Puts tracking script in the head instead of the body
-        head: false,
-        // Setting this parameter is optional
-        anonymize: true,
-        // Setting this parameter is also optional
-        respectDNT: true,
-      },
-    },
-  ],
-}
-```
+Variables are set when JavaScript is compiled so when the development server is started
+or you build your site.
 
-This will then load the values from the relevant environment's `.env.*` file and make them available via GraphQL queries and the analytics plugin respectively.
+## Add .env\* files to .gitignore
 
-Local testing of the `staging` environment can be done with:
+Environment variable files should not be committed to Git as they often contain secrets
+which are not safe to add to Git. Instead, add `.env.*` to your `.gitignore` file and
+setup the environment variables manually on Gatsby Cloud and locally.
 
-```shell
-GATSBY_ACTIVE_ENV=staging npm run develop
-```
+## Environment variables on Gatsby Cloud
 
-**Note:** The `GATSBY_ACTIVE_ENV` will not be set automatically. You have to explicitly set the environment variable (e.g. by adding it to the `gatsby` command or setting it inside your terminal and/or CI provider). If not set it'll be `undefined`.
+In Gatsby Cloud you can configure environment variables in each site's "Site Settings".
```

Comment:
I think the section on Staging, Test etc. should be included as well but can be more concise

Reply:
👍